core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestratorSpec.scala

```diff
@@ -373,6 +373,45 @@ class BillingProjectOrchestratorSpec extends AnyFlatSpec {
 
   }
 
+  it should "fail if preDeletionSteps throws an exception " in {
```

Comment:
Added a test point to enforce this behavior (that we don't delete the billing project in this case).